module.nix

```diff
@@ -5,49 +5,191 @@ self:
 , lib
 , pkgs
 , ...
-}: let
+}:
+
+with lib;
+
+let
   cfg = config.programs.gBar;
 
   defaultGBarPackage = self.defaultPackage.x86_64-linux;
 
 in {
   options.programs.gBar = {
-    enable = lib.mkEnableOption "Wether to enable gBar, a blazingly fast statusbar written in c++";
+    enable = mkEnableOption "Wether to enable gBar, a blazingly fast statusbar written in c++";
 
-    package = lib.mkOption {
-      type = lib.types.nullOr lib.types.package;
+    package = mkOption {
+      type = types.nullOr types.package;
       default = defaultGBarPackage;
-      defaultText = lib.literalExpression "<gBar flake>.packages.<system>.default";
-      example = lib.literalExpression "<gBar flake>.packages.<system>.default.override { }";
+      defaultText = literalExpression "<gBar flake>.packages.<system>.default";
+      example = literalExpression "<gBar flake>.packages.<system>.default.override { }";
       description = ''
         gBar package to use.
       '';
     };
 
-    extraConfig = lib.mkOption {
-      type = lib.types.nullOr lib.types.lines;
-      default = null;
+    config = mkOption {
+        default = {};
+        description = "Options to write to gBar config file, named as standard options";
+        type = types.submodule {
+            options = {
+                CPUThermalZone = mkOption {
+                    type = types.nullOr types.str;
+                    default = "/sys/devices/pci0000:00/0000:00:18.3/hwmon/hwmon2/temp1_input";
+                    description = "path to the cpu thermal sensor, probably something in /sys/device";
+                };
+                SuspendCommand = mkOption {
+                    type = types.str;
+                    default = "systemctl suspend";
+                    description = "";
```

Comment:
Can you give each option a small description (Like CPUThermalZone). It probably gives a bit more context and you can copy it from the default config.

Reply:
yep but, also should all options have set default values or should they just be empty?

Reply:
I think so. Even though gBar has hardcoded defaults for most configs and doesn't need them, setting them in module.nix too could make things easier to understand.